### ISSUE TITLE
[Download][Button][Accessibility] generated anchor/button mark-up does not comply with good accessibility

### ISSUE THREAD

**Issue by anneangersbach** [adobe/aem-core-wcm-components#1070]
## Feature Request

**Is your feature request related to a problem? Please describe.**
The download and button component produce an anchor tag. This is not compliant with WCAG (or their names - see issues: [905](https://github.com/adobe/aem-core-wcm-components/issues/905) and [737](https://github.com/adobe/aem-core-wcm-components/issues/737))

**Describe the solution you'd like**
A Button element would be a lot better suited for these components: “A button is a widget that enables users to trigger an action or event, such as submitting a form, opening a dialog, […]” [WCAG#Button](https://www.w3.org/TR/wai-aria-practices/#button)

I’d like to be able to define if the components generate a button or a link. This could be done in the dialog via a toggle - or by providing policy options, so these can be defined per use-case. These would also add the option of setting up aria-labels if desired.

**Are there alternatives?**
Overlaying the html, however this would not give the option of entering aria labels, and could potentially break compatibility to future component updates.

**Comment by gabrielwalt** [comment]
For the Button component, the author can already choose between a `<button>` and a `<a href="…">`, depending on whether the author provided a link URL or not: without a link, it renders a `<button>`; with a link, it renders a `<a href="…">`.

For the Download component, as it links to a file, it always renders as a `<a href="…">`. Rendering a `<button>` would make little sense as it would require additional JS to function properly.

When a link provides an `href` attribute, the element isn't strictly speaking a button anymore, but rather a call to action link as it redirects to another location when clicking it. Therefore, it shouldn't be added a `role="button"` attribute when it renders as a `<a href="…">`.

Also, `tabindex="0"` isn't necessary as this is only needed for links without an `href` attribute, or for other elements that are not a `<button>`.

The only missing accessibility feature that is currently missing is the `aria-pressed` option, but this would only make sense if the button is a toggle, which would need a corresponding configuration and setup, not only in the context of accessibility.

Therefore, I'll close this issue as won't fix. If some accessibility issues remain, please reopen a more specific issue about what exactly is missing.
